fix : Top Bar Titles updation

revanthkumarJ opened the pull request:
Fixes - [Jira-#231](https://mifosforge.jira.com/browse/MM-231)

Didn't create a Jira ticket, click [here](https://mifosforge.jira.com/jira/software/c/projects/MM/issues/) to create new.

Please Add Screenshots If there are any UI changes.

| Before                                     | After                                  |
|--------------------------------------------|----------------------------------------|
|                                            |                                        |


Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the static analysis check `./gradlew check` or `ci-prepush.sh` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

revanthkumarJ committed:
updated

revanthkumarJ committed:
updated TopBars

revanthkumarJ committed:
removed dependencies blocks

revanthkumarJ committed:
updated

HekmatullahAmin commented on feature/about/src/commonMain/kotlin/org/mifos/mobile/feature/about/ui/PrivacyPolicyScreen.kt:
```diff
@@ -40,10 +41,12 @@ internal fun PrivacyPolicyScreen(
         backPress = navigateBack,
         modifier = modifier,
         content = {
+                padding ->
```
Moved WebView into the trailing lambda block after MifosScaffold(...)
This is cleaner and more "Compose-style" when the last parameter is a composable content block. and also change padding to paddingValues

sth like this: 
```
 MifosScaffold(
        topBarTitle = stringResource(Res.string.feature_about_privacy_policy),
        backPress = navigateBack,
        modifier = modifier,
    ) { paddingValues ->
        WebView(
            url = stringResource(Res.string.feature_about_policy_url),
            isLoading = isLoading,
            onLoadingChange = { isLoading = it },
            modifier = Modifier.padding(paddingValues),
        )
    }
```

revanthkumarJ replied:
Done

revanthkumarJ committed:
updated

Nagarjuna0033 commented on core/network/build.gradle.kts:
```diff
@@ -59,14 +59,3 @@ kotlin {
     }
 }
-
-dependencies {
```
@revanthkumarJ This dependency block is included to enable KSP (Kotlin Symbol Processing). Any reason to remove it.

revanthkumarJ replied:
Usually when we convert to kmp and cmp we move the things to kotlin source sets so i thought it is also moved to otlin source sets

revanthkumarJ replied:
@Nagarjuna0033  can't we move them to kotlin sourceSets commonMain ?

Nagarjuna0033 replied:
> @Nagarjuna0033 can't we move them to kotlin sourceSets commonMain ?

No @revanthkumarJ Because KSP processes Kotlin code differently for each target as you can see there are jsMain, and other platforms so I believe we cant move it into commonMain

revanthkumarJ replied:
ok i understood now they can't me moved inside source sets moving them back

Nagarjuna0033 commented:
@revanthkumarJ Add a screenshot what you have refactored.

revanthkumarJ committed:
updated

revanthkumarJ commented:
@Nagarjuna0033 

In Privacy policy screen

| Before                                     | After                                  |
|--------------------------------------------|----------------------------------------|
|                          ![image](https://github.com/user-attachments/assets/8022ffd0-f6f8-45b9-be1b-4cf5421c125c)    |    ![image](https://github.com/user-attachments/assets/c7edb6f4-c820-4de8-90c4-6b8cb4ecbe1e)                                    |


And in other screens topBar i didn't changed ui .changed fontSizes only

Before:
uses for some topbars 24.sp for some titleLarge and some titleMedium

Now:
changed all to follow a similar one i.e titleMedium

niyajali commented on feature/guarantor/src/commonMain/kotlin/org/mifos/mobile/feature/guarantor/screens/guarantorDetails/GuarantorDetailTopBar.kt:
```diff
@@ -43,7 +43,12 @@ internal fun GuarantorDetailTopBar(
 
     TopAppBar(
```
The `TopAppBar` in this file and the below files could be replaced with the `MifosTopAppBar` and if needed some extra functionality on some screens, consider creating overload and using it throughout the app for better consistency and management.

revanthkumarJ replied:
done

niyajali reviewed (changes requested):
@revanthkumarJ We're creating TopAppBar for each screen instead of using the core `MifosTopAppBar` component which does the same. consider use that one

revanthkumarJ committed:
Replaced TopBars with MifosTopBar

revanthkumarJ committed:
Merge branch 'kmp-impl' into Enhancements

revanthkumarJ commented:
@niyajali @Nagarjuna0033 can you review it once and approve this if no changes are required

niyajali commented on core/ui/src/commonMain/kotlin/org/mifos/mobile/core/ui/component/UserProfileTopBar.kt:
```diff
@@ -9,33 +9,29 @@
     modifier: Modifier = Modifier,
 ) {
-    TopAppBar(
+    MifosTopBarTitleComposable(
```
Rename it to MifosTopAppBar

niyajali commented on core/ui/src/commonMain/kotlin/org/mifos/mobile/core/ui/component/UserProfileTopBar.kt:
```diff
@@ -9,33 +9,29 @@
 ) {
-    TopAppBar(
+    MifosTopBarTitleComposable(
         title = {
```
And the **title lamda** composable should only contain title(text) and the **icon** should be moved to **action** composable **lamba**

niyajali commented on feature/about/src/commonMain/kotlin/org/mifos/mobile/feature/about/ui/AboutUsScreen.kt:
```diff
@@ -55,14 +53,9 @@ internal fun AboutUsScreen(
     MifosScaffold(
         topBar = {
-            MifosTopBarTitleComposable(
+            MifosTopBarTitle(
```
Edit: - Rename it to **MifosTopAppBar** and create one more overload which takes the **navigation icon** and **actions** and replace it on above screen.

revanthkumarJ replied:
I didn't understand , can you somewhat elaborate.
I think there is a typo in "ove screen."

Nagarjuna0033 replied:
> I didn't understand , can you somewhat elaborate. I think there is a typo in "ove screen."

@revanthkumarJ He is saying What you have done in this like replacing TopBar with our customized MifosTopBar. Do this for overall project like where ever we have used TopBar just replace that with MifosTopBar by making adjustments if needed

niyajali replied:
@revanthkumarJ Create required overloads and used those all over the project and the overload names should be MifosTopAppBar

niyajali reviewed (changes requested):
Create these overload and use all over the project based on its requirement instead of creating **TopAppBar** for each screen

revanthkumarJ committed:
updated

revanthkumarJ committed:
updated

revanthkumarJ commented:
@niyajali removed all externally defined topBars and used only MifosTopAppBar and done necessary changes so that they work for all top Bars

can you review it once

niyajali reviewed (approved):
Have you verified that the `TopAppBar` maintains a consistent appearance and behavior across all screens in the application? Could you ensure it follows the same design guidelines (e.g., elevation, title style, navigation icon, and actions) on every screen where it's used?

revanthkumarJ commented:
> Have you verified that the `TopAppBar` maintains a consistent appearance and behavior across all screens in the application? Could you ensure it follows the same design guidelines (e.g., elevation, title style, navigation icon, and actions) on every screen where it's used?

@niyajali yes i have verified it in all screens the functionality is working fine and it followed same styles like fontSize medium and navigationIcon (backArrow in all screen and menu in home etc.)
